runtime-modules/membership/src/lib.rs

```diff
@@ -273,24 +209,23 @@ decl_module! {
             // make sure we are accepting new memberships
             ensure!(Self::new_memberships_allowed(), "new members not allowed");
 
-            // ensure paid_terms_id is active
-            let terms = Self::ensure_active_terms_id(paid_terms_id)?;
+            let fee = T::MembershipFee::get();
 
             // ensure enough free balance to cover terms fees
-            ensure!(T::Currency::can_slash(&who, terms.fee), "not enough balance to buy membership");
+            ensure!(T::Currency::can_slash(&who, fee), "not enough balance to buy membership");
```

Comment:
Are `can_slash` and `slash` correct to use? I know they were already there, but from the docs it says

`Deducts up to value from the combined balance of who, preferring to deduct from the free balance. This function cannot fail.`

So presumably `can_slash` will also count funds that are locked for some purpose, but we really don't want people to be able to use any locked funds for buying a membership. Only free balance should count, the handbook says

```
Free balance of the signer account exceeds `membership_price`. 
```

Reply:
Fixed:
- changed `free_balance` check with `usable_balance`
- added a test for the fix

Reply:
Excellent, and thanks for the offline explainer.